www/spatial-navigation.js

```diff
@@ -0,0 +1,269 @@
+// @ts-check
+export const CONTAINER_SELECTOR = '[data-spatial-navigation-contain]';
+export const FOCUSABLE_SELECTOR = '[tabindex], a[href], button:not([disabled])';
+export const DIRECTION_BY_KEY = {
+  ArrowDown: 'down',
+  ArrowLeft: 'left',
+  ArrowRight: 'right',
+  ArrowUp: 'up',
+};
+
+/**
+ * Enable spatial navigation with keyboard Arrow keys
+ */
+export function enableSpatialNavigation() {
+  document.addEventListener('keydown', handleKeyDown);
+}
+
+/**
+ * Disable spatial navigation
+ */
+export function disableSpatialNavigation() {
+  document.removeEventListener('keydown', handleKeyDown);
+}
+
+/**
+ * Handle keydown event
+ * @param {KeyboardEvent} event
+ */
+function handleKeyDown(event) {
+  // Bail if entering text into input/textarea
```

Comment:
entering text into input/textarea should not accidentally trigger a navigation

Reply:
Same as above.